Frontend assignment1

sangwan491 opened the pull request:
Please review and provide feedback.

<img width="1438" alt="image" src="https://github.com/user-attachments/assets/ed2ce96e-309a-416b-a737-9d207fcc58e1" />

sangwan491 committed:
completed assignment1

piyushsarin-sib commented on styles.css:
```diff
@@ -0,0 +1,135 @@
+  font-family: 'Inter', sans-serif;
+}
+
+.modal-header {
```
Pleas use css variable

piyushsarin-sib commented on styles.css:
```diff
@@ -0,0 +1,135 @@
+    text-align: center;
+}
+
+.card-container__auto {
```
You can also work on the responsiveness of your page using media queries

piyushsarin-sib commented on index.html:
```diff
@@ -0,0 +1,116 @@
+                    <div class="card-container">
+                        <div class="card-button">
+                            <div class="card-image">
+                                <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOcAAABUCAYAAAB5qiD7AAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAgHSURBVHgB7d19UFV1Hsfxz72XAownJRRhSUSEyFl0bRVbzQcgXZ7MFmhUMAVtQaedlrJyNoldEDQlhU0D0lADEUVFXBpbM3T/Wtf1oenBNlulVFSEELgEXO6Fu/fcGXem0Kir95zfoc9rhn84VzzO+J7z+/G9/NC09vaaQURCMZvNKVoQkZAYJ5GgGCeRoBgnkaAYJ5GgGCeRoBgnkaAYJ5GgGOc ...
```
Please prefer WebP image over encoded images

piyushsarin-sib commented on index.html:
```diff
@@ -0,0 +1,116 @@
+        <div class="modal">
+            <header class="modal-header">
+                <div class="modal-heading">Create a campaign</div>
+                <button id="close-campaign-modal-btn">X</button>
```
Please review the WCAG guidelines and apply appropriate aria-label attributes and semantic HTML tags wherever necessary.

piyushsarin-sib commented on index.html:
```diff
@@ -0,0 +1,116 @@
+        </div>
+    </div>
+    <script>
+        const campaignModal = document.getElementById('campaign-modal');
```
Please keep js in separate file

piyushsarin-sib commented on styles.css:
```diff
@@ -0,0 +1,135 @@
+    flex-wrap: wrap;
+}
+
+.card-button__auto {
```
Clickable elements should be applied with proper 
focus and animation styles

piyushsarin-sib commented on styles.css:
```diff
@@ -0,0 +1,135 @@
+  z-index: 999;
+}
+
+.modal {
```
The overflow: scroll in .modal can result in double scrollbars on some browsers. Use overflow-y: auto instead:


.modal {
    overflow-y: auto;
}

piyushsarin-sib commented on styles.css:
```diff
@@ -0,0 +1,135 @@
+  height: 100vh;
+  background-color: rgba(0, 0, 0, 0.3);
+  display: none;
+  z-index: 999;
```
You've used z-index: 999 and 1000. Ensure no conflict with other components. A good practice is to maintain a z-index scale in your design system (e.g., backdrop = 1000, modal = 1100, tooltip = 1200).

piyushsarin-sib commented on styles.css:
```diff
@@ -0,0 +1,135 @@
+
+.card-container__auto {
+    margin-top: 8px;
+    display: flex;
```
void redundancy in class names like .card-container and .card-container__auto. Consider naming based on functionality:

.card-container--auto instead of __auto, if it's a modifier.

Alternatively, apply BEM convention properly.

sangwan491 committed:
Applied transition and changed styles

sangwan491 committed:
Fix header div closure, update title font size, adjust padding and margin

- Fixed issue with improperly closed header div
- Changed font size for the title
- Adjusted padding and margin values for layout improvements

sangwan491 committed:
refactor: move inline JavaScript to separate index.js file

piyushsarin-sib commented on styles.css:
```diff
@@ -0,0 +1,297 @@
+
+/* ===== OPEN CAMPAIGN BUTTON ===== */
+#open-campaign-modal-btn {
+    margin-top: 8px;
```
FYI: We have naos design token for spacing as well: https://naos.brevo.design/@dtsl/docs-react/?path=/docs/guidelines-spacing--docs
